packages/babel-types/src/definitions/core.js

```diff
@@ -280,7 +280,9 @@ defineType("File", {
       validate: assertNodeType("Program"),
     },
     comments: {
-      validate: assertEach(assertNodeType("Comment")),
+      validate: !process.env.BABEL_TYPES_8_BREAKING
+        ? Object.assign(() => {}, { each: { type: "Comment" } })
+        : assertEach(assertNodeType("Comment")),
```

Comment:
one thing, I noticed, when looking into referred issue:

`assertNodeType("Comment")` results in correct typescript types, but actual check is not correct instead it should be `assertNodeType("CommentBlock", "CommentLine")`